### PR TITLE
fixed #34 run py90dot1 from the command line

### LINKED ISSUE #34: run py90dot1 from the command line
**Problem**: py90dot1 cannot be run from the command line
**Solution**: Have something like 

`python py90dot1.py proposed_idffile climatezone weatherfile`

that will output the baseline IDF

### PR DESCRIPTION
**Pull request by santoshphilip**
**Problem**: py90dot1 cannot be run from the command line
**Solution**: Have something like

python py90dot1.py proposed_idffile climatezone weatherfile

that will output the baseline IDF

Also added command line interface for all the modules. So you can run any of them through the command line

### CODE REVIEW

**Comment by pyenergyplus** [comment]
@ncbrown , @jamiebull1 , Can one of you merge this into the master. It may trigger merge conflicts if it is not merged early enough.

File: py90dot1/baselinesystem.py [hunk 1]
```diff
@@ -1,7 +1,41 @@
-# Copyright (c) 2017 Santosh Philip
+# Copyright (c) 2017-1018 Santosh Philip
```
Comment (jamiebull1) [inline py90dot1/baselinesystem.py]: 2018

Comment (santoshphilip) [reply py90dot1/baselinesystem.py]: Good point. That was a copy paste from the tutorial. I'll take a look and cleanup clean up as we continue

**Comment by santoshphilip** [comment]
There is a merge conflict because issue #32 got merged before this.
I'll fix the merge conflict and update the pull request

**Comment by santoshphilip** [comment]
fixed the merge conflicts.

- Copyright dates have been fixed
- I am leaving the repetitive decorators as they are
     - I want to leave it explicit as to how it follows from the click tutorial
- test have to yet be written for the command line stuff
- Will open another issue to for the command line tests.

**Comment by santoshphilip** [comment]
This merge has to be done quickly or rejected (it is 6 days pending. C4 contract asks for a 2 day merge)
I am OK with a rejection. It will let us figure out how to handle a rejection in the C4 contract.

**Comment by jamiebull1** [comment]
Well that's another thing I don't like about the contract then :-) For me there's nothing wrong with a merge request staying open as long as there are outstanding change requests. But fine about the decorator, just wanted to suggest it.

If you're happy without tests, fine with me.

**Comment by pyenergyplus** [comment]
I am up at 1am, just as your emails come in. 

Here is my understanding of the C4 contract.

1. Not having the test for the command line is just me getting sloppy and lazy. You would have been right to reject it for that reason, and hold me to that standard.
2. The comment about the decorator can be done by opening a new issue.

We are fleshing out our understanding of the contract as we work on this. Right now we are all known collaborators working on this project. The C4 mechanism allows a stranger to contribute to the software with the least friction.

**Comment by pyenergyplus** [comment]
Thinking this thru, I am of the opinion that you should reject the patch for not having tests.
Is it possible to roll back at this point ?

**Comment by pyenergyplus** [comment]
I am going copy paste a section of the C4 contract here, so it becomes part of the record of this discussion.

### 2.3. Patch Requirements

1. Maintainers and Contributors MUST have a Platform account and SHOULD use their real names or a well-known alias.
2. A patch SHOULD be a minimal and accurate answer to exactly one identified and agreed problem.
3. A patch MUST adhere to the code style guidelines of the project if these are defined.
4. A patch MUST adhere to the "Evolution of Public Contracts" guidelines defined below.
5. A patch SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
6. A patch MUST compile cleanly and pass project self-tests on at least the principle target platform.
7. A patch commit message MUST consist of a single short (less than 50 characters) line stating the problem ("Problem: ...") being solved, followed by a blank line and then the proposed solution ("Solution: ...").
8. A "Correct Patch" is one that satisfies the above requirements.

**Comment by santoshphilip** [comment]
I am up at 1am, just as your emails come in.

Here is my understanding of the C4 contract.

1. Not having the test for the command line is just me getting sloppy and lazy. You would have been right to reject it for that reason, and hold me to that standard.
2. The comment about the decorator can be done by opening a new issue.

We are fleshing out our understanding of the contract as we work on this. Right now we are all known collaborators working on this project. The C4 mechanism allows a stranger to contribute to the software with the least friction.

**Comment by santoshphilip** [comment]
Thinking this thru, I am of the opinion that you should reject the patch for not having tests.
Is it possible to roll back at this point ?

**Comment by santoshphilip** [comment]
I am going copy paste a section of the C4 contract here, so it becomes part of the record of this discussion.

### 2.3. Patch Requirements

1. Maintainers and Contributors MUST have a Platform account and SHOULD use their real names or a well-known alias.
2. A patch SHOULD be a minimal and accurate answer to exactly one identified and agreed problem.
3. A patch MUST adhere to the code style guidelines of the project if these are defined.
4. A patch MUST adhere to the "Evolution of Public Contracts" guidelines defined below.
5. A patch SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
6. A patch MUST compile cleanly and pass project self-tests on at least the principle target platform.
7. A patch commit message MUST consist of a single short (less than 50 characters) line stating the problem ("Problem: ...") being solved, followed by a blank line and then the proposed solution ("Solution: ...").
8. A "Correct Patch" is one that satisfies the above requirements.

**Comment by jamiebull1** [comment]
Off topic, but where would one submit a patch to the C4 contract? Pretty sure they mean 'principal' not 'principle'. I don't see a github repo anywhere.

**Comment by jamiebull1** [comment]
And I don't think I can easily roll back the accepted pull request. You could open a new PR from the previous state and I'll accept it, but seems like mindless rule-following for little or no benefit.

**Comment by santoshphilip** [comment]
I found this on C4
https://github.com/csirtgadgets/C4

it has not been updated in a while. 
Not sure who else maintains C4  it after Hintjens passed away. in 2016.

**Comment by santoshphilip** [comment]
OK let us not roll back. I'll open a new issue on testing and clean up.

**Pull request merged**